src/platform/packages/shared/kbn-esql-ast/src/mutate/commands/rerank/index.ts

```diff
@@ -55,17 +59,77 @@ export const setFields = (
   cmd.fields.push(...(fields as ESQLAstRerankCommand['fields']));
 };
 
-export const setInferenceId = (cmd: ESQLAstRerankCommand, id: string | ESQLIdentifierOrParam) => {
-  if (typeof id === 'string') {
-    id = id[0] === '?' ? Builder.param.build(id) : Builder.identifier(id);
-  }
 
-  if (id.type !== 'identifier' && id.type !== 'literal') {
-    throw new Error(`Invalid RERANK inferenceId: ${id}`);
+export const setWithParameter = (
+  cmd: ESQLAstRerankCommand,
+  key: string,
+  value: string | ESQLStringLiteral | ESQLParamLiteral
```

Comment:
The function adds or update only one parameter at a time. It's not very smart, but I don't know how perfect we want to make it.